### PR TITLE
Issue 624 - addCollectionGroupSnapshotListener

### LINKED ISSUE #624: feat(firestore): add addCollectionGroupSnapshotListener
### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [ ] Authentication
- [ ] Crashlytics
- [X] Cloud Firestore
- [ ] Cloud Messaging
- [ ] Cloud Storage
- [ ] Performance
- [ ] Remote Config

### Current problem

I am not sure but I think there is a very important feature missing. What I am trying to do is addCollectionSnapshotListener on collectionGroup query.
Such a thing can be done with javascript firebase library like this:
`const resources = query(collectionGroup(getFirestore(), 'resources'));`
`onSnapshot(resources, (querySnapshot) => {`
`  querySnapshot.forEach((doc) => {`
`    console.log(doc.data());`
`  });`
`});`

### Preferred solution

Since there already are [getCollection](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore#getcollection), [addCollectionSnapshotListener](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore#adddocumentsnapshotlistener) and [getCollectionGroup](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore#getcollectiongroup) functions in API, I would suggest addCollectionGroupSnapshotListener function is added.

I made a quick glance at [addCollectionSnapshotListener web code](https://github.com/capawesome-team/capacitor-firebase/blob/792c4d15bc6f1a57ffa41d6eaa5d2436f56f9fb3/packages/firestore/src/web.ts#L204) and I think whole function can be copied and only change that is needed is on [line 212](https://github.com/capawesome-team/capacitor-firebase/blob/792c4d15bc6f1a57ffa41d6eaa5d2436f56f9fb3/packages/firestore/src/web.ts#L212) where `'collecion'` should be changed to `'collectionGroup'`.

### Alternative options

_No response_

### Additional context

_No response_

### Before submitting

- [X] I have read and followed the [feature request guidelines](https://capawesome.io/contributing/feature-requests/).
- [ ] I have attached links to possibly related issues and discussions.

### PR DESCRIPTION
**Pull request by svrznjak**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] The changes have been tested successfully. - **Only tested on Web and Android**
- [X] A changeset has been created (`npm run changeset`).
- [X] I have read and followed the [pull request guidelines](https://capawesome.io/contributing/pull-requests/).

Close #624

### CODE REVIEW

File: packages/firestore/README.md
Comment (robingenz) [inline packages/firestore/README.md:1]: Please revert the unnecessary changes in this file. Also, make sure to run `nm run docgen`.

File: .changeset/cold-zebras-pump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@capacitor-firebase/firestore': minor
+---
+
+Added addCollectionGroupSnapshotListener to support snapshot listeners on collectionGroup query.
```
Comment (robingenz) [inline .changeset/cold-zebras-pump.md:5]:
```suggestion
feat: add `addCollectionGroupSnapshotListener` method
```

File: packages/firestore/ios/Plugin/Classes/Options/AddCollectionGroupSnapshotListenerOptions.swift
Comment (robingenz) [inline packages/firestore/ios/Plugin/Classes/Options/AddCollectionGroupSnapshotListenerOptions.swift:1]: You must add this file via XCode in order to register the file properly.

File: packages/firestore/ios/Plugin/FirebaseFirestorePlugin.swift [hunk 2]
```diff
@@ -265,6 +265,41 @@ public class FirebaseFirestorePlugin: CAPPlugin {
         }
     }
 
+    @objc func addCollectionGroupSnapshotListener(_ call: CAPPluginCall) {
+        call.keepAlive = true
+
+        guard let reference = call.getString("reference") else {
+            call.reject("collectionId must be provided.")
```
Comment (robingenz) [inline packages/firestore/ios/Plugin/FirebaseFirestorePlugin.swift:272]:
```suggestion
            call.reject(errorReferenceMissing)
```

File: packages/firestore/src/definitions.ts [hunk 3]
```diff
@@ -84,6 +84,15 @@ export interface FirebaseFirestorePlugin {
     options: AddCollectionSnapshotListenerOptions,
     callback: AddCollectionSnapshotListenerCallback<T>,
   ): Promise<CallbackId>;
+  /**
+   * Adds a listener for collection group snapshot events.
+   *
+   * @since TODO
+   */
+  addCollectionGroupSnapshotListener<T extends DocumentData = DocumentData>(
+    options: GetCollectionGroupOptions,
```
Comment (robingenz) [inline packages/firestore/src/definitions.ts:93]:
```suggestion
    options: AddCollectionGroupSnapshotListenerOptions,
```

File: packages/firestore/src/definitions.ts [hunk 4]
```diff
@@ -334,6 +343,30 @@ export interface AddCollectionSnapshotListenerOptions {
   queryConstraints?: QueryNonFilterConstraint[];
 }
 
+/**
+ * @since TODO
```
Comment (robingenz) [inline packages/firestore/src/definitions.ts:347]:
```suggestion
 * @since 6.1.0
```

**Review by robingenz — CHANGES_REQUESTED** [review]

File: packages/firestore/README.md
Comment (svrznjak) [reply packages/firestore/README.md:1]: Oh, I did not notice prettier made changes before I committed changes. Now I reverted README.md, added an example of addCollectionGroupSnapshotListener, and then ran `npm run docgen`.

**Review by robingenz — APPROVED** [review]

**Comment by robingenz** [comment]
Please fix the reported errors.

**Pull request closed without merging**